oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/server/ServerBearerTokenAuthenticationConverter.java

```diff
@@ -90,8 +91,20 @@ public void setAllowUriQueryParameter(boolean allowUriQueryParameter) {
 		this.allowUriQueryParameter = allowUriQueryParameter;
 	}
 
-	private static String resolveFromAuthorizationHeader(HttpHeaders headers) {
-		String authorization = headers.getFirst(HttpHeaders.AUTHORIZATION);
+	/**
+	 * Set this value to configure what header is checked when resolving a Bearer Token.
+	 * This value is defaulted to {@link HttpHeaders#AUTHORIZATION}.
+	 *
+	 * This allows other headers to be used such as {@link HttpHeaders#PROXY_AUTHORIZATION}
+	 *
+	 * @param authorizationHeader the header to check when retrieving the Bearer Token.
+	 */
+	public void setAuthorizationHeader(String authorizationHeader) {
```

Comment:
I'm thinking that you aren't precisely setting the header, but actually the header name. Also, it's not the authorization header if we are changing it, right? 

Could we instead call this `setBearerTokenHeaderName`?